app/views/courses/CoursesUpdateAccountView.coffee

```diff
@@ -0,0 +1,72 @@
+errors = require 'core/errors'
+RootView = require 'views/core/RootView'
+template = require 'templates/courses/courses-update-account-view'
+AuthModal = require 'views/core/AuthModal'
+{logoutUser, me} = require('core/auth')
+
+module.exports = class CoursesUpdateAccountView extends RootView
+  id: 'courses-update-account-view'
+  template: template
+
+  events:
+    'click .login-btn': 'onClickLogInButton'
+    'click .logout-btn': 'onClickLogoutButton'
+    'click .remain-teacher-btn': 'onClickRemainTeacherButton'
+    'click .update-teacher-btn': 'onClickUpdateTeacherButton'
+    'click .remain-student-btn': 'onClickRemainStudentButton'
+    'click .update-student-btn': 'onClickUpdateStudentButton'
+
+  initialize: (options) ->
+    @accountType = switch
+      when me.isTeacher() then $.i18n.t('courses.teacher')
+      when me.isStudent() then $.i18n.t('courses.student')
+
+  onClickLogInButton: ->
+    @openModalView(new AuthModal())
+    application.tracker?.trackEvent 'Started Student Login', category: 'Courses Update Account'
+
+  onClickLogoutButton: ->
+    Backbone.Mediator.publish("auth:logging-out", {})
+    logoutUser()
+
+  onClickRemainTeacherButton: (e) ->
+    @remainTeacher(e.target, 'Remain teacher')
+
+  onClickUpdateTeacherButton: (e) ->
+    $(e.target).prop('disabled', true)
+    application.tracker?.trackEvent 'Update teacher', category: 'Courses Update Account'
+    application.router.navigate('/teachers/update-account', {trigger: true})
+
+  onClickRemainStudentButton: (e) ->
+    @becomeStudent(e.target, 'Remain student')
+
+  onClickUpdateStudentButton: ->
+    return unless window.confirm($.i18n.t('courses.update_account_confirm_update_student'))
+    @becomeStudent(e.target, 'Update student')
+  
+  becomeStudent: (targetElem, trackEventMsg) ->
+    $(targetElem).prop('disabled', true)
+    $.ajax({
+      url: '/db/user/-/become-student',
+      type: 'PUT',
+      success: ->
+        application.tracker?.trackEvent trackEventMsg, category: 'Courses Update Account'
+        application.router.navigate('/courses', {trigger: true})
+      error: ->
+        $(targetElem).prop('disabled', false)
+        errors.showNotyNetworkError(arguments...)
+    })
+
+  remainTeacher: (targetElem, trackEventMsg) ->
+    $(targetElem).prop('disabled', true)
+    $.ajax({
+      url: '/db/user/-/remain-teacher'
+      type: 'PUT'
+      success: ->
+        application.tracker?.trackEvent trackEventMsg, category: 'Courses Update Account'
+        application.router.navigate('/teachers', {trigger: true})
+      error: ->
+        $(targetElem).prop('disabled', false)
+        errors.showNotyNetworkError(arguments...)
+    })
```

Comment:
We should avoid using `$.ajax` in views, instead going through Backbone Collections and Models. The Model or Collection to use is whatever is returned. In this case, the endpoints don't return anything... I think it would make the most sense to return the User which has just been modified. So this would be in `User`:

```
  remainTeacher: (options={}) ->
    options.url = '/db/user/-/remain-teacher'
    options.type = 'PUT'
    @fetch(options)
```

And in the view:

```
  @remainTeacher({
    success: ->
      application.tracker?.trackEvent trackEventMsg, category: 'Courses Update Account'
      application.router.navigate('/teachers', {trigger: true})
    error: ->
      $(targetElem).prop('disabled', false)
      errors.showNotyNetworkError(arguments...)
  })
```

That way API details are not the view's concern, and in addition, `me` gets the latest `role`.